### PR TITLE
Fix mov_j timeout because of the flange orientation

### LINKED ISSUE #145: Action 'mov_j' execution times-out in most cases
**Describe the bug**
mov_j action successful when specifying `position: {x: 0.36, y: 0, z: 0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}`, but it times-out when specifying `position: {x: 0, y: 0.36, z: 0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}`. Other positions such as `position: {x: 0.2, y: 0.2, z: 0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}` or other positions fail, too.

**To Reproduce**
1. run `ros2 launch mg400_bringup mg400.launch.py`
2. run `ros2 action send_goal /mov_j mg400_msgs/action/MovJ "{pose: {header: {frame_id: mg400_origin_link, stamp: {sec: 0, nanosec: 0}}, pose: {position: {x: 0, y: 0.36, z: 0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}}}"`
3. The arm moves to the specified position, but it doesn't return result for a while, and then mg400 node says `execution timeout`

**Expected behavior**
After moving the arm, it returns true as a result.

### PR DESCRIPTION
**Pull request by ynyBonfennil**
This pull request fixes two points

1. compare `tf2::getYaw(pose.orientation)` and `tf2::getYaw(goal.orientation)` to evaluate how much the current flange orientation is distant from the goal.
2. fix orientation value of the current flange
    - As I wrote on the code, `RealtimeFeedbackTcpInterface::getCurrentEndPose` returns the position of `mg400_origin_link` but its orientation is actually based on `mg400_end_effector_flange`. Hence I've converted the value from that of `mg400_end_effector_flange` to `mg400_origin_link`.

### CODE REVIEW

File: mg400_plugin/src/motion_api/mov_j.cpp [hunk 1]
```diff
@@ -104,7 +104,9 @@ void MovJ::execute(const std::shared_ptr<GoalHandle> goal_handle)
       return is_in_tolerance(pose.position.x - goal.position.x, tolerance_mm) &&
              is_in_tolerance(pose.position.y - goal.position.y, tolerance_mm) &&
              is_in_tolerance(pose.position.z - goal.position.z, tolerance_mm) &&
-             is_in_tolerance(pose.orientation.w - goal.orientation.w, tolerance_rad);
+             is_in_tolerance(
+        tf2::getYaw(pose.orientation) - tf2::getYaw(goal.orientation),
```
Comment (ynyBonfennil) [inline mg400_plugin/src/motion_api/mov_j.cpp:108]: It seems like `Uncrustify` requires these lines to have this indentation (if not, the test fails)

**Comment by m12watanabe1a** [comment]
@ynyBonfennil Thank you so much for the change.

I think it is actually the bug of getEndPose method. 
Could you try this PR instead?

https://github.com/HarvestX/MG400_ROS2/pull/148

**Comment by ynyBonfennil** [comment]
I'm not sure if #148 have some effect on other MG400_ROS2 features, but it solved the problem. I've removed the code of converting the value of orientation.

**Comment by m12watanabe1a** [comment]
> I'm not sure if #148 have some effect on other MG400_ROS2 features, but it solved the problem. I've removed the code of converting the value of orientation.

I think the method linked below was givening you a wrong EndPoint position (wichi is not considered about the flange angle.) before you merge #148
https://github.com/HarvestX/MG400_ROS2/blob/f365461ee58334f029fad080f07f908b0868db37/mg400_plugin/src/motion_api/mov_j.cpp#L117

**Comment by ynyBonfennil** [comment]
I remember that the original mg400 (like MovJ, MovL etc.) protocol doesn't care about the flange's orientation,is it correct? If we follow the rule, then the previous getCurrentEndPose implementation might have a reason to some extent...
But anyway, cosistent frame id in a single message is more of the ROS2 style, so the new implementation would be suitable for MG400_ROS2.

**Comment by m12watanabe1a** [comment]
> But anyway, cosistent frame id in a single message is more of the ROS2 style, so the new implementation would be suitable for MG400_ROS2.

Yes.. I want to follow the ROS2-style.

Thank you so much for the consideraion.

**Review by m12watanabe1a — APPROVED** [review]
LGTM

**Pull request merged**